openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -1211,8 +1275,11 @@ def test_fetching_program_discounted_price_as_anonymous_user(self):
             body=json.dumps(mock_discount_data),
             content_type='application/json'
         )
+        user = AnonymousUserFactory()
+        user.courseenrollment_set = CourseEnrollment.objects.none()
+        user.courseentitlement_set = CourseEntitlement.objects.none()
```

Comment:
Is `courseentitlement_set` a magic method of some sort? I can't find where this gets set.

Reply:
Honestly, in my understanding it is a magic method, as is `courseenrollment_set`. I just know that, when used, they respectively provide the set of all course entitlements a user holds and the set of all course enrollments a user holds.